<?xml version="1.0"?>
<launch>
  <arg name="robot_ip" doc="IP of controller" />
  <arg name="robot_ns" default=""/>
  <arg name="use_ns" default="0"/>
  

  <param name="robot_ip_address" type="str" value="$(arg robot_ip)" unless="$(arg use_ns)"/>
  <param name="$(arg robot_ns)/robot_ip_address" type="str" value="$(arg robot_ip)" if="$(arg use_ns)"/>

  <param name="rcs_port" type="int" value="5181"/>
  
  <node name="motion_service" pkg="lebai_driver" type="motion_service"   unless="$(arg use_ns)" output="screen">
  </node>


  <node name="$(arg robot_ns)_motion_service" pkg="lebai_driver" type="motion_service" if="$(arg use_ns)" output="screen">
    <remap from="controller_joint_names" to="$(arg robot_ns)/controller_joint_names"/>
    <remap from="robot_ip_address" to="$(arg robot_ns)/robot_ip_address"/>
    <remap from="joint_trajectory_action" to="$(arg robot_ns)/joint_trajectory_action"/>
    <remap from="joint_path_command" to="$(arg robot_ns)/joint_path_command"/>
    <remap from="robot_status" to="$(arg robot_ns)/robot_status"/>
    <remap from="stop_motion" to="$(arg robot_ns)/stop_motion"/>
  </node>



</launch>